### PR TITLE
Validate double literals

### LINKED ISSUE #19: Parser does not validate double literals
The parser should produce an error [FOAR0002] if numeric overflow or underflow occurs when parsing a double literal.

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #19

### CODE REVIEW

**Comment by sharwell** [comment]
|  | Tests Run | Failures | Errors | Skipped |
| --- | --- | --- | --- | --- |
| Before | 13414 | 644 | 226 | 0 |
| After | 13414 | 641 | 226 | 0 |

File: bundles/org.eclipse.wst.xml.xpath2.processor/src/org/eclipse/wst/xml/xpath2/processor/internal/XPathFlex.java [hunk 1]
```diff
@@ -1317,7 +1318,7 @@ else if (zzAtEOF) {
           }
         case 170: break;
         case 49: 
-          { return symbol(XpathSym.DOUBLE, new Double(yytext()));
+          { return symbol(XpathSym.DOUBLE, LiteralUtils.parseDouble(yytext()));
```
Comment (RackerWilliams) [inline bundles/org.eclipse.wst.xml.xpath2.processor/src/org/eclipse/wst/xml/xpath2/processor/internal/XPathFlex.java]: I believe you are putting the modification on the wrong file here.  Should be in https://github.com/sharwell/webtools.sourceediting.xpath/blob/75f20765480c1704043ef68dc59d0d5db5bf603b/bundles/org.eclipse.wst.xml.xpath2.processor/grammars/xpath.lex#L180

Comment (sharwell) [inline bundles/org.eclipse.wst.xml.xpath2.processor/src/org/eclipse/wst/xml/xpath2/processor/internal/XPathFlex.java]: I'll submit a fix for this after #42 is merged.

**Comment by sharwell** [comment]
Ready for review again!

**Comment by RackerWilliams** [comment]
Looks good 👍

**Pull request merged**